app/javascript/packs/suspendUserModal.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable no-restricted-globals */
+import {
+  closeWindowModal,
+  showWindowModal,
+  WINDOW_MODAL_ID,
+} from '@utilities/showModal';
+import { request } from '@utilities/http';
+
+const suspendOrUnsuspendUser = async ({
+  event,
+  btnAction,
+  userId,
+  suspendOrUnsuspendReason,
+}) => {
+  event.preventDefault();
+  closeModal();
+
+  try {
+    const response = await request(
+      `/admin/member_manager/users/${userId}/user_status`,
+      {
+        method: 'PATCH',
+        body: JSON.stringify({
+          id: userId,
+          user: {
+            note_for_current_role: suspendOrUnsuspendReason,
+            user_status: btnAction == 'suspend' ? 'Suspended' : 'Good standing',
+          },
+        }),
+        credentials: 'same-origin',
+      },
+    );
+
+    const outcome = await response.json();
+
+    if (outcome.success) {
+      top.addSnackbarItem({
+        message: outcome.message,
+        addCloseButton: true,
+      });
+    } else {
+      top.addSnackbarItem({
+        message: 'Error: something went wrong.',
+        addCloseButton: true,
+      });
+    }
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+};
+
+function closeModal() {
+  closeWindowModal(window.parent.document);
+}
+
+const modalContents = new Map();
+
+/**
+ * Helper function to handle finding and caching modal content. Since our Preact modal helper works by duplicating HTML content,
+ * and our modals rely on IDs to label form controls, we remove the original hidden content from the DOM to avoid ID conflicts.
+ *
+ * @param {string} modalContentSelector The CSS selector used to identify the correct modal content
+ */
+function getModalContents(modalContentSelector) {
+  if (!modalContents.has(modalContentSelector)) {
+    const modalContentElement =
+      window.parent.document.querySelector(modalContentSelector);
+    const modalContent = modalContentElement.innerHTML;
+
+    modalContentElement.remove();
+    modalContents.set(modalContentSelector, modalContent);
+  }
+
+  return modalContents.get(modalContentSelector);
+}
```

Comment:
(suggestion / not blocking) We only use this file for the one modal right? If so, I think we can simplify this a touch as we don't need to maintain a map (since there's only ever one)

```suggestion
let modalContents;

/**
 * Helper function to handle finding and caching modal content. Since our Preact modal helper works by duplicating HTML content,
 * and our modals rely on IDs to label form controls, we remove the original hidden content from the DOM to avoid ID conflicts.
 *
 * @param {string} modalContentSelector The CSS selector used to identify the correct modal content
 */
function getModalContents(modalContentSelector) {
  if(!modalContents) {
    const modalContentElement =
        window.parent.document.querySelector(modalContentSelector);
     modalContents = modalContentElement.innerHTML;
     modalContentElement.remove();
  }
  return modalContents;
}
```